Firely.Fhir.Packages/Core/FhirCiClient.cs

```diff
@@ -0,0 +1,1083 @@
+        }
+
+        /// <summary>
+        /// Downloads the current qas.json file from the build server and deserializes into an array of <see cref="FhirCiQaRecord"/>.
+        /// </summary>
+        /// <returns>An array of FhirCiQaRecord objects representing the current FhirCiQaRecords.</returns>
+        private async Task<List<FhirCiQaRecord>?> downloadGuideQAs()
+        {
+            // download the QA records from the build server
+            HttpRequestMessage request = new HttpRequestMessage()
+            {
+                Method = HttpMethod.Get,
+                RequestUri = _qasUri,
+                Headers =
+                {
+                    Accept =
+                    {
+                        new MediaTypeWithQualityHeaderValue("application/json"),
+                    },
+                },
+            };
+
+            HttpResponseMessage response = await _httpClient.SendAsync(request);
+            System.Net.HttpStatusCode statusCode = response.StatusCode;
+
+            if (statusCode != System.Net.HttpStatusCode.OK)
+            {
+                return null;
+            }
+
+            string json = await response.Content.ReadAsStringAsync();
+            if (string.IsNullOrEmpty(json))
+            {
+                return null;
+            }
+
+            List<FhirCiQaRecord>? qas = JsonConvert.DeserializeObject<List<FhirCiQaRecord>>(json);
+
+            // if we do not have records, we are done
+            if (qas == null)
+            {
+                return null;
+            }
+
+            return qas;
+        }
+
+        private async Task<List<FhirCiQaRecord>> downloadCoreQAs()
+        {
+            List<FhirCiQaRecord> qas = new();
+
+            // download the branch list from the core build
+            HttpRequestMessage request = new HttpRequestMessage()
+            {
+                Method = HttpMethod.Get,
+                RequestUri = _ciBranchUri,
+                Headers =
+                {
+                    Accept =
+                    {
+                        new MediaTypeWithQualityHeaderValue("application/json"),
+                    },
+                },
+            };
+
+            HttpResponseMessage response = await _httpClient.SendAsync(request);
+            System.Net.HttpStatusCode statusCode = response.StatusCode;
+
+            if (statusCode != System.Net.HttpStatusCode.OK)
+            {
+                return qas;
+            }
+
+            string json = await response.Content.ReadAsStringAsync();
+            if (string.IsNullOrEmpty(json))
+            {
+                return qas;
+            }
+
+            CiBranchRecord[]? coreBranches = JsonConvert.DeserializeObject<CiBranchRecord[]>(json);
+
+            if (coreBranches == null)
+            {
+                return qas;
+            }
+
+            // traverse the core branches to build their records
+            foreach (CiBranchRecord ciBranchRec in coreBranches)
```

Comment:
This looks like a separate function to me

Reply:
Fair, will split those apart.